tools/validators/instance_validator/main.py

```diff
@@ -12,40 +12,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import instance_parser 
-import sys
+"""Main of the instance validator."""
+
+from __future__ import print_function
+
+import instance_parser
+import argparse
 # ONTOLOGY VALIDATION
 # import ontology_validation
 # import generate_universe
 
 # TODO add input and return type checks in all functions
 
 if __name__ == '__main__':
-    if len(sys.argv) != 2:
-        print('USAGE: python3 schema.py yaml_file_path')
-        sys.exit(1)
-    
-    # SYNTAX VALIDATION
-    print('\nValidator starting ...')
-    filename = sys.argv[1]
+  parser = argparse.ArgumentParser(
+      description='Validate a YAML building configuration file')
+  parser.add_argument('-i', '--input',
+                      dest='filename',
+                      required=True,
+                      help='Filepath to YAML building configuration',
+                      metavar='FILE')
+  arg = parser.parse_args()
+
+  # SYNTAX VALIDATION
+  print('\nValidator starting ...')
+  filename = arg.filename
 
-    # throws errors for syntax
-    parsed = dict(instance_parser.parse_yaml(filename))
+  # throws errors for syntax
+  parsed = dict(instance_parser.parse_yaml(filename))
 
-    print('Passed syntax checks!')
+  print('Passed syntax checks!')
 
-    # ONTOLOGY VALIDATION
-    '''
-    print('Building ontology universe ...')
+  # ONTOLOGY VALIDATION
+  '''
+  print('Building ontology universe ...')
 
-    universe = generate_universe.build_universe()
-    fields, subfields_map, states_map, units_map, entities_map = generate_universe.parse_universe(universe)
+  universe = generate_universe.build_universe()
+  parsed_univ = generate_universe.parse_universe(universe)
+  fields, subfields_map, states_map, units_map, entities_map = parsed_univ
```

Comment:
if you're returning 5 values, it's much more readable to create a data structure to contain them.  We often use named tuples for this (find examples in the ontology validator)

Reply:
Okay, I'll fix this in the ontology generation PR and make a TODO to change these lines once every module is merged

Reply:
create an issue and reference it in your todo please as follows TODO(url): replace this assignment logic with NamedTuple ontology generation

Reply:
nice !